### ISSUE TITLE
💅 useAwaitThenable: false positive calling a Promise-returning property of an intersection type (2.5.7+)

### ISSUE THREAD

**Issue by liorp** [biomejs/biome#11359]
### Environment information

```block
CLI:
  Version:                      2.5.8

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_DISTRIBUTION:           npm
  BIOME_CONFIG_PATH:            unset
  JS_RUNTIME_VERSION:           v26.7.0
  JS_RUNTIME_NAME:              node
```

### Rule name

`nursery/useAwaitThenable`

### Summary

When a function type is reached through a **property of an intersection object
type**, `useAwaitThenable` reports `await` on it as "await used on a non-Promise
value", even though the property's declared return type is `Promise<void>`.

This is a separate defect from #11083. There the *return type itself* was the
intersection (`Promise<T> & { cancel(): void }`). Here the return type is a plain
`Promise<void>`, and the intersection is on the **object that owns the property**.
I verified the two are independent below: the #11083 reproduction is fixed in
2.5.7, while this one still reports on 2.5.7 and 2.5.8.

### Reproduction

`biome.json`:

```json
{
  "linter": {
    "enabled": true,
    "domains": { "project": "recommended", "types": "recommended" },
    "rules": {
      "preset": "recommended",
      "nursery": { "useAwaitThenable": "error" }
    }
  }
}
```

`src/case1.ts`:

```ts
type Hook = (input: { readonly id: string }) => Promise<void>;
type Base = { readonly name: string };
type Deps = Base & { readonly hook?: Hook };

// REPORTED, but `hook` is declared `=> Promise<void>`.
export async function optionalCall(deps: Deps): Promise<void> {
	await deps.hook?.({ id: "x" });
}

// REPORTED, and here `deps.hook` is narrowed to `Hook`.
export async function guardedCall(deps: Deps): Promise<void> {
	if (deps.hook === undefined) return;
	await deps.hook({ id: "x" });
}

// NOT reported - correct. Plain parameter, no intersection.
export async function plainCall(hook: Hook): Promise<void> {
	await hook({ id: "x" });
}
```

`biome check src/case1.ts`:

```block
src/case1.ts:7:2  lint/nursery/useAwaitThenable
src/case1.ts:13:2 lint/nursery/useAwaitThenable
```

`plainCall` is not reported, so the trigger is reaching the function type through
a property of the intersection, not the `await` itself. Removing `Base &` from
`Deps` makes both diagnostics disappear.

### Version behaviour

Same file, same config, only the version changed:

| Version | `optionalCall` + `guardedCall` (this issue) | #11083 reproduction |
| --- | --- | --- |
| 2.5.4 | not reported | not reported |
| 2.5.5 | not reported | reported |
| 2.5.7 | **reported** | not reported (fixed) |
| 2.5.8 | **reported** | not reported (fixed) |

So the #11083 fix landed, and this variant is still open. It first appears in
2.5.7.

### Expected behaviour

Neither `optionalCall` nor `guardedCall` should be reported. `Hook` returns
`Promise<void>`, so both `await`s are on a real promise.

### Why this one bites

The suggested action is to remove the `await`. On the real code this pattern came
from, that turns an awaited call into a floating promise, and in one case fed a
pending `Promise` into a Zod `.parse()` call. `noFloatingPromises` would then
report the very line `useAwaitThenable` asked us to change, so the two rules
disagree about the same expression.

The pattern is ordinary dependency injection: a deps object composed with `&`,
carrying optional async seams.

```ts
type RouteDeps = AuthDeps & { readonly beginMission?: BeginMission };
```

### Playground link

Not provided: the rule is type-aware and I did not want to link a playground
result that may look clean for the wrong reason. The reproduction above is one
self-contained file with no imports.

**Comment by ematipico** [comment]
Please, next time, share a reproduction. Here [one](https://biomejs.dev/playground/?lintRules=useAwaitThenable&tab=formatter&pane=Diagnostics&code=dAB5AHAAZQAgAEgAbwBvAGsAIAA9ACAAKABpAG4AcAB1AHQAOgAgAHsAIAByAGUAYQBkAG8AbgBsAHkAIABpAGQAOgAgAHMAdAByAGkAbgBnACAAfQApACAAPQA%2BACAAUAByAG8AbQBpAHMAZQA8AHYAbwBpAGQAPgA7AAoAdAB5AHAAZQAgAEIAYQBzAGUAIAA9ACAAewAgAHIAZQBhAGQAbwBuAGwAeQAgAG4AYQBtAGUAOgAgAHMAdAByAGkAbgBnACAAfQA7AAoAdAB5AHAAZQAgAEQAZQBwAHMAIAA9ACAAQgBhAHMAZQAgACYAIAB7ACAAcgBlAGEAZABvAG4AbAB5ACAAaABvAG8AawA%2FADoAIABIAG8AbwBrACAAfQA7AAoACgAvAC8AIABSAEUAUABPAFIAVABFAEQALAAgAGIAdQB0ACAAYABoAG8AbwBrAGAAIABpAHMAIABkAGUAYwBsAGEAcgBlAGQAIABgAD0APgAgAFAAcgBvAG0AaQBzAGUAPAB2AG8AaQBkAD4AYAAuAAoAZQB4AHAAbwByAHQAIABhAHMAeQBuAGMAIABmAHUAbgBjAHQAaQBvAG4AIABvAHAAdABpAG8AbgBhAGwAQwBhAGwAbAAoAGQAZQBwAHMAOgAgAEQAZQBwAHMAKQA6ACAAUAByAG8AbQBpAHMAZQA8AHYAbwBpAGQAPgAgAHsACgAJAGEAdwBhAGkAdAAgAGQAZQBwAHMALgBoAG8AbwBrAD8ALgAoAHsAIABpAGQAOgAgACIAeAAiACAAfQApADsACgB9AAoACgAvAC8AIABSAEUAUABPAFIAVABFAEQALAAgAGEAbgBkACAAaABlAHIAZQAgAGAAZABlAHAAcwAuAGgAbwBvAGsAYAAgAGkAcwAgAG4AYQByAHIAbwB3AGUAZAAgAHQAbwAgAGAASABvAG8AawBgAC4ACgBlAHgAcABvAHIAdAAgAGEAcwB5AG4AYwAgAGYAdQBuAGMAdABpAG8AbgAgAGcAdQBhAHIAZABlAGQAQwBhAGwAbAAoAGQAZQBwAHMAOgAgAEQAZQBwAHMAKQA6ACAAUAByAG8AbQBpAHMAZQA8AHYAbwBpAGQAPgAgAHsACgAJAGkAZgAgACgAZABlAHAAcwAuAGgAbwBvAGsAIAA9AD0APQAgAHUAbgBkAGUAZgBpAG4AZQBkACkAIAByAGUAdAB1AHIAbgA7AAoACQBhAHcAYQBpAHQAIABkAGUAcABzAC4AaABvAG8AawAoAHsAIABpAGQAOgAgACIAeAAiACAAfQApADsACgB9AAoACgAvAC8AIABOAE8AVAAgAHIAZQBwAG8AcgB0AGUAZAAgAC0AIABjAG8AcgByAGUAYwB0AC4AIABQAGwAYQBpAG4AIABwAGEAcgBhAG0AZQB0AGUAcgAsACAAbgBvACAAaQBuAHQAZQByAHMAZQBjAHQAaQBvAG4ALgAKAGUAeABwAG8AcgB0ACAAYQBzAHkAbgBjACAAZgB1AG4AYwB0AGkAbwBuACAAcABsAGEAaQBuAEMAYQBsAGwAKABoAG8AbwBrADoAIABIAG8AbwBrACkAOgAgAFAAcgBvAG0AaQBzAGUAPAB2AG8AaQBkAD4AIAB7AAoACQBhAHcAYQBpAHQAIABoAG8AbwBrACgAewAgAGkAZAA6ACAAIgB4ACIAIAB9ACkAOwAKAH0A)
